core/src/main/java/bisq/core/btc/wallet/WalletService.java

```diff
@@ -408,18 +430,15 @@ public TransactionConfidence getConfidenceForTxId(String txId) {
         return null;
     }
 
-    protected TransactionConfidence getTransactionConfidence(Transaction tx, Address address) {
-        List<TransactionConfidence> transactionConfidenceList = getOutputsWithConnectedOutputs(tx)
-                .stream()
-                .filter(WalletService::isOutputScriptConvertibleToAddress)
-                .filter(output -> address != null && address.equals(getAddressFromOutput(output)))
-                .map(o -> tx.getConfidence())
-                .collect(Collectors.toList());
-        return getMostRecentConfidence(transactionConfidenceList);
+    private TransactionConfidence getTransactionConfidence(Transaction tx, Address address) {
+        boolean matchesAddress = getOutputsWithConnectedOutputs(tx).stream()
+                .anyMatch(output -> address != null && address.equals(getAddressFromOutput(output)));
+
+        return matchesAddress ? getMostRecentConfidence(List.of(tx.getConfidence())) : null;
```

Comment:
Yes, I wasn't completely sure of the intent of the original version and tried to preserve its behaviour, but it looks like it may have been broken. The original stream pipeline had the line:

> .map(o -> tx.getConfidence())

But `tx` is the underlying method parameter and so this just fed a repeating list of the same `TransactionConfidence` object into `getMostRecentConfidence`.

I think it may be fixed by restoring the original pipeline, but using the lines:

> .flatMap(o -> Stream.ofNullable(o.getParentTransaction()))
> .map(Transaction::getConfidence)

in place of the `o -> tx.getConfidence` mapping.

This would make the method return null if (there was no matching output and) none of the tx inputs with the given address has a connecting parent tx (though I'm not sure if the latter ever happens). I guess that would be the right behaviour.

Reply:
Oh, yes looks like that `.map(o -> tx.getConfidence())` was a bug. I think intention was to find all txs where the address was used for receiving or sending BTC. And take the most recent confidence. Unforunately the design flaw to use addresses so dominantely causes many issues in many area. Maybe we can slowely try to convert the domain model to use txos as the core entity and if the UI needs to show addresses we grab the collection of txos related to that address.